src/processors/tone/ladder_filter/HighPassOnePole.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+// One-pole high-pass filter modeled by the TPT approach.
+// Used in the transistor-ladder filter cascade.
+// (See Zavalishin "The Art of VA Filter Design")
+class HighPassOnePole
+{
+public:
+    HighPassOnePole() = default;
+    ~HighPassOnePole() = default;
+
+    // Initialize filter
+    void reset (const double _sample_rate);
+
+    double get_sample_rate() const;
+
+    //==============================================================================
+
+    // Set the "instantaneous gain" of the filter
+    // Indirectly sets the filter cutoff frequency
+    // (G is calculated when setting the cutoff of the full ladder structure)
+    void set_G (const double _G);
+
+    // Set the "instantaneous gain" of the trapezoidal integrator times two
+    // (g2 is calculated when setting the cutoff of the full ladder structure)
+    void set_g2 (const double _g2);
+
+    // Get current contents of the internal state register
+    double get_state() const;
+
+    //==============================================================================
+
+    // Process a single audio sample
+    double process (double x);
```

Comment:
I wonder if we might benefit from some of these per-sample processing functions being marked `inline`... Compilers these days are probably smart enough to inline them anyway, but it might be worth a try.

Reply:
Consider this done.